src/borrow/mod.rs

```diff
@@ -533,20 +536,19 @@ where
     /// });
     /// ```
     pub fn resize<ID: IntoDimension>(self, dims: ID) -> PyResult<Self> {
-        let array = self.array;
-
         // SAFETY: Ownership of `self` proves exclusive access to the interior of the array.
         unsafe {
-            array.resize(dims)?;
+            self.array.resize(dims)?;
         }
 
+        let array = self.array.clone();
```

Comment:
Hhhmmm, this reference count bump was not necessary before. Can we rewrite this really pass on ownership of `self.array`? I think `drop` here is just a short-hand for calling `release` and hence should be avoidable.

Reply:
Can we just do this? We skip both `acquire_mut` from the constructor and `release_mut` from `Drop`

```rust
pub fn resize<ID: IntoDimension>(self, dims: ID) -> PyResult<Self> {
    // SAFETY: Ownership of `self` proves exclusive access to the interior of the array.
    unsafe {
        self.array.resize(dims)?;
    }

    Ok(self)
}
```

Reply:
I don't think so because the old borrow contains metadata on its shape and needs to be updated after the resize. This could theoretically done in place, but that is much more complicated than acquire/release and furthermore would require an additional method in the capsule-based API which supports cross-extension borrow tracking. Long story short, we do need to release/acquire pair.

Reply:
I see, in that case can we just insert a release/acquire pair before the return?

```rust
...

release_mut(self.array.py(), self.array.as_array_ptr());
acquire_mut(self.array.py(), self.array.as_array_ptr())?;

Ok(self)
```